### PR TITLE
feat: disablePayloadLogs flag to suppress payload info

### LINKED ISSUE #212: Actual Payload Shared in Step Functions Console On Function Error
# Problem

I want to use AWS Lambda Power Tuning to invoke Lambda functions with an input payload that contains credentials. I do not want this information in the AWS Step Functions execution history.

Initially I explored the use of a pre-processor to workaround this by allowing the pre-processor to fetch credentials and pass those direct to the Lambda being tuned. This I understand would all execute within a single state of the Step Functions, so credentials would not be shared in the Step Functions console.

However, if the Lambda being tuned throws an error, we [log the payload](https://github.com/alexcasalboni/aws-lambda-power-tuning/blob/4d0d35d0410550574d23931d3cc13a0e32dabd0e/lambda/executor.js#L162C110-L162C142), which in my case would include the credentials.

# Proposed Solutions

Two ideas spring to mind, but I'm open to discussing the problem further and open to other suggestions how to solve this:

1. we stop including the payload in the error message thrown. The payload I think should be the same for each execution of the Step Functions, unless we use a pre-processor which returns different output given the same input. So is logging the payload valuable?

2. we introduce a new toggle to enable / disable inclusion of the payload in the error thrown, defaulting to showing the payload to maintain consistent behaviour, but this would allow users like me to disable it if desired.

# Contributing

I'm happy to contribute to this project with any agreed changes.

### PR DESCRIPTION
**Pull request by rrhodes**
# What?

New execution parameter to indicate whether input payloads should be suppressed from error messages and logs.

# Why

Closes #212.

### CODE REVIEW

File: README-SAR.md [hunk 1]
```diff
@@ -53,6 +53,8 @@ The state machine accepts the following input parameters:
...
 * **postProcessorARN** (string): it must be the ARN of a Lambda function; if provided, the function will be invoked after every invocation of `lambdaARN`; more details below in the Pre/Post-processing functions section
 * **discardTopBottom** (number between 0.0 and 0.4, by default is 0.2): By default, the state machine will discard the top/bottom 20% of "outliers" (the fastest and slowest), to filter out the effects of cold starts that would bias the overall averages. You can customize this parameter by providing a value between 0 and 0.4, with 0 meaning no results are discarded and 0.4 meaning that 40% of the top/bottom results are discarded (i.e. only 20% of the results are considered).
+* **sleepBetweenRunsMs** (integer) If provided, the time in milliseconds that the tuner function will sleep/wait after invoking your function, but before carrying out the Post-Processing step, should that be provided. This could be used if you have aggressive downstream rate limits you need to respect. By default this will be set to 0 and the function won't sleep between invocations. Setting this value will have no effect if running the invocations in parallel.
```
Comment (rrhodes) [inline README-SAR.md:56]: Parameter `sleepBetweenRunsMs` was missing here, although all other parameters appeared to be shared from what is detailed in README-INPUT-OUTPUT, so I think this is supposed to be documented here as well?

Happy to revert this addition if that's not the case.

File: README-INPUT-OUTPUT.md [hunk 2]
```diff
@@ -21,7 +21,8 @@ The state machine accepts the following input parameters:
...
-* **sleepBetweenRunsMs** (integer) If provided, the time in milliseconds that the tuner function will sleep/wait after invoking your function, but before carrying out the Post-Processing step, should that be provided. This could be used if you have agressive downstream rate limits you need to respect. By default this will be set to 0 and the function won't sleep between invocations. Setting this value will have no effect if running the invocations in parallel.
+* **sleepBetweenRunsMs** (integer) If provided, the time in milliseconds that the tuner function will sleep/wait after invoking your function, but before carrying out the Post-Processing step, should that be provided. This could be used if you have aggressive downstream rate limits you need to respect. By default this will be set to 0 and the function won't sleep between invocations. Setting this value will have no effect if running the invocations in parallel.
```
Comment (rrhodes) [inline README-INPUT-OUTPUT.md:24]: Typo fix: `agressive` -> `aggressive`

File: test/unit/test-utils.js [hunk 3]
```diff
@@ -890,4 +906,51 @@ describe('Lambda Utils', () => {
...
+
+            expect(consoleLogArg).to.contain('Invoking function');
+            expect(consoleLogArg.includes('with payload')).to.be(isPayloadInConsoleLog);
+        };
+
+        before(() => {
+            if (consoleLogSetupStub) {
+                consoleLogStub = consoleLogSetupStub;
+            } else {
+                consoleLogStub = sinon.stub(console, 'log');
```
Comment (alexcasalboni) [inline test/unit/test-utils.js:929]: Why is this needed here?

**Review by alexcasalboni — CHANGES_REQUESTED** [review]
Looks great 🚀  Only a couple of minor comments/doubts.

File: test/unit/test-lambda.js [hunk 4]
```diff
@@ -45,7 +45,7 @@ const invokeForSuccess = async(handler, event) => {
     }
 };
 
-// utility to invoke handler (success case)
+// utility to invoke handler (failure case)
```
[the remark this answers is not in the document]
Comment (rrhodes) [reply test/unit/test-lambda.js:48]: Ah, I misread this, I'll update the comment to articulate what you've said above, thanks. 👍🏻

File: test/unit/test-utils.js (hunk 3 above)
Comment (rrhodes) [reply test/unit/test-utils.js:929]: This is a workaround because of how we [setup tests](https://github.com/alexcasalboni/aws-lambda-power-tuning/blob/master/test/setup.spec.js#L9) with sinon stubs.

I wanted to keep this setup where we could choose to comment out the setup stubs if desired, without breaking this specific test logic.

Open to suggestions if we want to revisit how tests are set up now, or if we want to improve how we handle stubs in this specific test case.

Comment (alexcasalboni) [reply test/unit/test-utils.js:929]: Got it! I think that's fine for now :)

File: lambda/executor.js [hunk 5]
```diff
@@ -133,17 +135,22 @@ const extractDataFromInput = async(event) => {
         postProcessorARN: input.postProcessorARN,
         discardTopBottom: discardTopBottom,
         sleepBetweenRunsMs: sleepBetweenRunsMs,
+        disablePayloadLogs: input.disablePayloadLogs,
```
[the remark this answers is not in the document]
Comment (rrhodes) [reply lambda/executor.js]: Added: [cbd86a6](https://github.com/alexcasalboni/aws-lambda-power-tuning/pull/213/commits/cbd86a63b77fe247fec508eacd2f92866faa02fa)

**Comment by alexcasalboni** [comment]
Thank you again, @rrhodes 🙏  do you use SAR for deployments? do you need this on SAR as a new app version asap?

**Comment by alexcasalboni** [comment]
@rrhodes done 😄  the new version `4.3.2` is available on SAR 🚀

**Comment by rrhodes** [comment]
Awesome, thank you @alexcasalboni! 😄
